libphal/temporary_file.H

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include <filesystem>
+#include <utility>
+
+namespace libphal
+{
+namespace util
+{
+
+namespace fs = std::filesystem;
+
+/**
+ * @class TemporaryFile
+ *
+ * A temporary file in the file system.
+ *
+ * The temporary file is created by the constructor.  The absolute path to the
+ * file can be obtained using getPath().
+ *
+ * Note: Callers responsibility to delete the file after usage.
+ *       The temporary file can be deleted by calling remove().
+ *
+ * TemporaryFile objects cannot be copied, but they can be moved.  This enables
+ * them to be stored in containers like std::vector.
+ */
+class TemporaryFile
+{
+       public:
+	// Specify which compiler-generated methods we want
+	TemporaryFile(const TemporaryFile &) = delete;
+	TemporaryFile &operator=(const TemporaryFile &) = delete;
```

Comment:
If we define move constructor of assignment operator these methods are implicitly deleted by the compiler no need to delete again.

if you choose to delete then add this too move constructor
TemporaryFile(TemporaryFile&& rhs) = delete;

Tested with example
note: ‘TemporaryFile::TemporaryFile(const TemporaryFile&)’ is implicitly declared as deleted because ‘TemporaryFile’ declares a move constructor or move assignment operator

Reply:
#include <iostream>
#include <string>
#include <memory>
using namespace std;

class TemporaryFile
{
  public:
    TemporaryFile(const std::string& path):path(path)
    {
        std::cout << "came into constructor" << std::endl;
    }

    //delete all copy and assignment operators
    TemporaryFile(TemporaryFile&& rhs) = delete;
    TemporaryFile& operator=(TemporaryFile&& rhs) = delete;

    ~TemporaryFile()
    {
        std::cout << "came into destructor" << std::endl;
    }
  private:
    std::string path;

};

void test(std::shared_ptr<TemporaryFile>& test)
{
    std::cout << "came into test method " << std::endl;
}

int main()
{
    std::shared_ptr<TemporaryFile> f1  = std::make_shared<TemporaryFile>("devender");
    test(f1);
    return 0;
}


simpler approach

Reply:
why force the user to call removeFile() also , there is a risk of temporary files being created if removeFile is not called. 

better apporach will be to use shared_ptr approach, this way we can delete the file in the destructor and even though shared_ptr is passed across only one delete will be called.